modules/ethereum/src/lib.rs

```diff
@@ -83,6 +99,33 @@ pub struct StoredHeader<Submitter> {
 	/// this is the set that has produced the block itself.
 	/// The hash is the hash of block where validators set has been enacted.
 	pub next_validators_set_id: u64,
+	/// Hash of the last block which has **SCHEDULED** validators set change.
+	/// Note that signal doesn't mean that the set has been (or ever will be) enacted.
+	/// Note that the header may already be pruned.
+	pub last_signal_block: Option<H256>,
+}
+
+/// Validators set as it is stored in the runtime storage.
+#[derive(Encode, Decode, PartialEq, RuntimeDebug)]
+#[cfg_attr(test, derive(Clone))]
+pub struct ValidatorsSet {
+	/// Validators of this set.
+	pub validators: Vec<Address>,
+	/// Hash of the block where this set has been signalled. None if this is the first set.
+	pub signal_block: Option<H256>,
+	/// Hash of the block where this set has been enacted.
+	pub enact_block: H256,
+}
+
+/// Validators set change as it is stored in the runtime storage.
```

Comment:
Looks like this was copied from the struct above :P

Reply:
It is `Validators set change` and struct above is `Validators set`.